tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -36,6 +36,72 @@
     ASSERT_TRUE([backingElement isKindOfClass:[WXFrameworkElement class]]);
 }
 
+@interface UIDeallocTestButton : UIButton {
+    NSCondition* _condition;
+    BOOL* _signaled;
+}
+@end
+
+@implementation UIDeallocTestButton
+
+- (void)setDeallocEvent:(NSCondition*)condition signaledFlag:(BOOL*)signaled {
+    _condition = condition;
+    _signaled = signaled;
+}
+
+-(void)dealloc {
+    [_condition lock];
+    *_signaled = YES;
+    [_condition signal];
+    [_condition unlock];
+    [super dealloc];
+}
+@end
+
+TEST(UIButton, CheckForLeaks) {
+    Microsoft::WRL::WeakRef weakXamlElement;
+    {
```

Comment:
>{ [](start = 4, length = 1)

Does scoping it like this mean all objects are freed at the end or could we still have some autoreleasepools that need to release objects? Just asking because that was one of the concerns with fixing #1757.